plugins/woocommerce/src/StoreApi/Routes/V1/Agentic/CheckoutSessions.php

```diff
@@ -136,6 +136,9 @@ protected function requires_nonce( \WP_REST_Request $request ) {
 	protected function get_route_post_response( \WP_REST_Request $request ) {
 		$checkout_session = new AgenticCheckoutSession( $this->cart_controller->get_cart_instance() );
 
+		// Set the session_id early so that it can be checked if the current request is agentic commerce.
+		AgenticCheckoutUtils::get_or_set_checkout_session_id();
```

Comment:
Can we store the result of this call within the session object? I think it's under-utilized, and we could use it to both add structure, and become slightly less reliant on the WC session.

```suggestion
		$checkout_session->set_id( AgenticCheckoutUtils::get_or_set_checkout_session_id() );
```

Reply:
Good reminder about `AgenticCheckoutSession`! 

I've gone further in f36a9b47c386b16e8044b99e9446889f4befaaad, and moved `get_or_set_checkout_session_id` as a part of `AgenticCheckoutSession` and in the constructor of `AgenticCheckoutSession`, we will use it. 

This is even more ideal as we hide the logic of generating checkout session ID within `AgenticCheckoutSession`. The consumers of `AgenticCheckoutSession` is just need to know about the checkout session ID.